README.md

```diff
@@ -1,2 +1,310 @@
+    return connector.handleLambdaCallback(evt, context, callback);
+};
+```
+
+## Proactive state callbacks
+
+Sensors and devices that can be controlled other than through the SmartThings mobile app can change state at any time.
+To ensure that the SmartThings platform is made aware of these state changes right away callsbacks can be implemented
+to call into the SmartThings cloud. These callbacks are secured via a token exchange dependent on the client ID
+and secret defined for the ST Schema connector in the Developer Workspace. The following example is a
+minimal implementation of a connector that supports these callback. It builds on the previous example by implementing
+the callbacks and exposing a web-service endpoint for executing device commands.
+
+#### app.js
+
+The connector app is now initialized with the ST Schema connector's client ID and secret, which are available from
+the Developer workspace. It also declares an `accessTokens` map to contain the list of connectors that need to be
+called when device state changes. Note that this simple implementation stores the connectors in memory, so restarting
+the sever will cause them to be lost. The app also has new `callbackAccessHandler` and `integrationDeletedHandler`
+handlers defined to add and remove entries from the `accessTokens` map.
+```javascript
+const {SchemaConnector} = require('st-schema')
+const deviceStates = {switch: 'off', level: 100}
+const accessTokens = {}
+const connector = new SchemaConnector()
+  .clientId(process.env.ST_CLIENT_ID)
+  .clientSecret(process.env.ST_CLIENT_SECRET)
+  .discoveryHandler((accessToken, response) => {
+    response.addDevice('external-device-1', 'Test Dimmer', 'c2c-dimmer')
+  })
+  .stateRefreshHandler((accessToken, response) => {
+    response.addDevice('external-device-1', [
+      {
+        component: 'main',
+        capability: 'st.switch',
+        attribute: 'switch',
+        value: deviceStates.switch
+      },
+      {
+        component: 'main',
+        capability: 'st.switchLevel',
+        attribute: 'level',
+        value: deviceStates.level
+      }
+    ])
+  })
+  .commandHandler((accessToken, response, devices) => {
+    for (const device of devices) {
+      response.addDevice(device.externalDeviceId, device.commands.map(cmd => {
+        if (cmd.command === 'setLevel') {
+          deviceStates.level = cmd.arguments[0];
+          return {
+            component: cmd.component,
+            capability: cmd.capability,
+            attribute: 'level',
+            value: deviceStates.level
+          }
+        } else {
+          deviceStates.switch = cmd.command === 'on' ? 'on' : 'off';
+          return {
+            component: cmd.component,
+            capability: cmd.capability,
+            attribute: 'switch',
+            value: deviceStates.switch
+          }
+        }
+      }))
+    }
+  })
+  .callbackAccessHandler(async (accessToken, callbackAuthentication, callbackUrls) => {
```

Comment:
All other handlers are not an async function except this one.  That can create confusion for the SDK user.

Reply:
:thumbsup: